README.md

```diff
@@ -512,6 +518,27 @@ The `React\Promise\PromisorInterface` provides a common interface for objects
 that provide a promise. `React\Promise\Deferred` implements it, but since it
 is part of the public API anyone can implement it.
 
+### AsyncInterop\Promise compatibility
+
+The [`PromiseInterface`](#promiseinterface) extends the
+[`AsyncInterop\Promise`](https://github.com/async-interop/promise) interface and
+thereby all promise implementations from React/Promise implement its `when()`
+method.
+
+The `AsyncInterop\Promise` interface is intended for interoperability with other
+libraries and their combinator and conversion functions.
+
+Note, that although the interface is named *Promise*, it has nothing to do with
+promises commonly known from the
+[CommonJS Promises/A](http://wiki.commonjs.org/wiki/Promises/A),
+[Promises/A+](https://promisesaplus.com) or
+[ECMAScript 2015 (ES6)](http://www.ecma-international.org/ecma-262/6.0/#sec-promise-constructor)
+specifications.
+
+If you're using React/Promise as your primary value placeholder implementation, 
+it is recommended to **not** use the `when()` method directly but the methods
+provided by the [`PromiseInterface`](#promiseinterface).
```

Comment:
It totally depends on what's consumer for you. Is it just application developers or also library developers? The latter will probably want to use `when()` if they also accept other promise implementations instead of wrapping all promises in React's promises. If all you need is a simple callback on completion (no chains), `when()` is pretty equivalent to `always()` / `done()` (depending whether you need the value / reason or not), except that it works on all compatible implementations then, not only React.

In fact, you might want to adapt `done()` to `when()` semantics regarding the error handling to provide a clear separation of concerns. IMO, throwing from `resolve` / `reject` because the consumption of the value failed is wrong. @jsor mentioned that in our face2face meeting as one of the ugliest things in the current promise implementation.

```
<?php

require __DIR__ . "/vendor/autoload.php";

$deferred = new React\Promise\Deferred;

$deferred->promise()->done(function ($value) {
    var_dump("value", $value);
}, function ($reason) {
    throw new Exception;
    var_dump("reason", $reason);
});

try {
    // shouldn't throw here, instead forward to the interop error handler
    $deferred->reject("foobar");
} catch (Exception $e) {
    var_dump("catch", $e);
}
```

Sorry for the lengthy discussion here, should I open a separate issue for that one? It depends on this PR, but I think it could make sense for 3.0.